File: 53_MaximumSubarray/maxSubArray.go
import "math"

func maxSubArray(nums []int) int {
    curSum, maxSum := math.MinInt, math.MinInt
    for _, num := range nums {
        if curSum < 0 {
            curSum = num
        } else {
            curSum += num
        }

        // go语言中没有:?三元表达式
        // go中Max, Min函数是针对float64类型数据的
        if maxSum < curSum {
            maxSum = curSum
        }
    }

    return maxSum
}

File: 53_MaximumSubarray/maxSubArray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        cur_sum, max_sum = float('-inf'), float('-inf')
        for num in nums:
            if cur_sum < 0:
                cur_sum = num
            else:
                cur_sum += num
            max_sum = max(max_sum, cur_sum)
        return max_sum
